### PR TITLE
Only update lastMessageText from pusher event from other users

### PR DESCRIPTION
**Pull request by tgolen**
cc @marcaaron 

### Details
This will do a better job of keeping the last comment text in the LHN in sync.

- It adds updates the text when the optimistic comment is first added to Onyx
- It only updates the text when the pusher event comes from the non-active chat participant

### Fixed Issues
<Please replace GH_LINK with the link to the GitHub issue this Pull Request is fixing>
Fixes https://github.com/Expensify/Expensify/issues/157488

### Tests
You need to have two chat clients running with two different accounts (use incognito or a different platform with a different user logged in)
1. Open a chat with both users
2. Make comments AS FAST AS YOU CAN by one user
3. Ensure the last message text is always up to date in the LHN
4. Make comments AS FAST AS YOU CAN by _the other_ user
3. Ensure the last message text is always up to date in the LHN

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web

image

#### Mobile Web

image
image

#### Desktop

image

#### iOS

image
image

#### Android

image
image

### CODE REVIEW

**Comment by francoisl** [comment]
Looks like tests didn't even start when GH Actions was down? I can't seem to manually retrigger them either, maybe you'll need to make a blank commit?

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -257,14 +257,21 @@ function updateReportWithNewAction(reportID, reportAction) {
...
         maxSequenceNumber: reportAction.sequenceNumber,
-        lastMessageTimestamp: reportAction.timestamp,
-        lastMessageText: messageText,
-        lastActorEmail: reportAction.actorEmail,
-    });
+    };
+
+    // If the report action from pusher is a higher sequence number than we know about (meaning it has come from
+    // a chat participant in another application), then the last message text and author needs to be updated as well
+    if (newMaxSequenceNumber > (lastReadSequenceNumbers[reportID] || 0)) {
```
Comment (marcaaron) [inline src/libs/actions/Report.js:268]: NAB, we could maybe do a 

```js
const lastReadSequenceNumber = lastReadSequenceNumbers[reportID] || 0;
```
above the `const updatedAction = {` line and use it here and when calculating the `unreadActionCount`

File: src/libs/actions/Report.js [hunk 2]
```diff
@@ -257,14 +257,21 @@ function updateReportWithNewAction(reportID, reportAction) {
     // Always merge the reportID into Onyx
     // If the report doesn't exist in Onyx yet, then all the rest of the data will be filled out
     // by handleReportChanged
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+    const updatedAction = {
```
Comment (marcaaron) [inline src/libs/actions/Report.js:260]: `updatedReportObject` ?

**Review by marcaaron — COMMENTED** [review]
LGTM only have a couple of small comments

**Comment by marcaaron** [comment]
Works great! 🎉

**Comment by tgolen** [comment]
Updated! Tests should run now, hopefully.

**Review by francoisl — APPROVED** [review]
Works well

**Review by marcaaron — APPROVED** [review]
🎉

**Pull request merged**